### PR TITLE
[WIP]Refactor vit using transformer.py in mmcv

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi,
  This pr refactored vision transformer using base classes in mmcv. As some necessary changes were made  in transformer.py mmcv, so all the unitests failed. And we'll fix ASAP.
Best regards
whcao

### CODE REVIEW

File: mmcls/models/backbones/vision_transformer.py [hunk 1]
```diff
@@ -1,251 +1,123 @@
...
+            operation_order=operation_order,
+            ffn_dropout=ffn_dropout,
+            act_cfg=act_cfg,
+            norm_cfg=norm_cfg,
+            ffn_num_fcs=ffn_num_fcs,
+            **kwargs)
+        assert len(self.operation_order) == 4
+        assert set(self.operation_order) == set(['self_attn', 'norm', 'ffn'])
+
+    # TODO: wait for the new version of mmcv, the current mmcv does not
```
Comment (jshilong) [inline mmcls/models/backbones/vision_transformer.py:59]: remove these comments

File: mmcls/models/backbones/vision_transformer.py [hunk 2]
```diff
@@ -1,251 +1,123 @@
...
+            Default：None
+        act_cfg (dict): The activation config for FFNs.
+        norm_cfg (dict): Config dict for normalization layer.
+        ffn_num_fcs (int): The number of fully-connected layers in FFNs.
+            Default：2.
     """
 
     def __init__(self,
-                 embed_dims,
+                 attn_cfgs,
```
Comment (jshilong) [inline mmcls/models/backbones/vision_transformer.py:39]:
```
                 *args,
                 ffn_dropout=0.0,
                 act_cfg=dict(type='GELU'),
                 **kwargs
``` 
may be better, you check the default value of ffn_dropout in mmcv

File: mmcls/models/backbones/vision_transformer.py [hunk 3]
```diff
@@ -1,251 +1,123 @@
...
-            `nn.MultiheadAttention`.
-        attn_drop (float): A Dropout layer on attn_output_weights. Default 0.0.
-        proj_drop (float): The drop out rate after attention. Default 0.0.
-    """
+        super(VitTransformerEncoderLayer, self).init_weights()
+        for ffn in self.ffns:
+            for m in ffn.modules():
+                if isinstance(m, nn.Linear):
+                    nn.init.xavier_normal_(m.weight)
+                    nn.init.normal_(m.bias, std=1e-6)
```
Comment (jshilong) [inline mmcls/models/backbones/vision_transformer.py:68]: self._is_init =True

**Comment by ycxioooong** [comment]
Close this PR since there is a new coming one for ViT.

**Comment by ycxioooong** [comment]
And my suggestion is that we create a totally new PR for refactoring ViT since the current commits are messy and have lots of distracting information.

**Pull request closed without merging**